milight_ibox2/milight_ibox2_control.py

```diff
@@ -135,18 +188,18 @@ def connect(self):
                 else:
                     print("TX retry %d..." % retry)
             self.socket_send(cmd_start_session)
-            data = self.socket_recv()
-            if data:
-                if len(data) != 22:
+            rx_data, (rx_addr, rx_port) = self.socket_recv()
```

Comment:
also here unpacking might fail if None is received

Reply:
I agree, and I've pushed a fix to this branch.